### ISSUE TITLE
keyToDegree is not the inverse of degreeToKey

### ISSUE THREAD

**Issue by shimpe** [supercollider/supercollider#3145]
58.keyToDegree(Scale.major, 12).degreeToKey(Scale.major,12)

-> 54
I expected 58 instead. 

Problem may be caused by:

58.keyToDegree(Scale.major, 12)

-> 33.5 

However, I expected 33.1 here, since degrees normally use the convention of adding .1 to add a semitone. This hypothesis is further confirmed by:

(58.keyToDegree(Scale.major, 12) - 0.4).degreeToKey(Scale.major,12)
-> 58

**Comment by shimpe** [comment]
forgot to mention: using sc 3.9dev

**Comment by brianlheim** [comment]
@jamshark70 @telephon do you think this could be changed in 3.9/3.10? or would it be better to document the current behavior. i know there has been some discussion about redoing degree/key things.

**Comment by jamshark70** [comment]
FWIW I just ran into this issue with my harmony processor and ended up rewriting the implementation in my own library:

https://github.com/jamshark70/ddwCommon/blob/master/Misc/ModalSpec.sc#L74-L80

AFAICS this is a correct inverse operation of degreeToKey:

```
// uses ddwCommon and ddwChucklib

(0..11).do { |key|
	var degree = key.mapMode(\default, true);  // cmaj
	[key, degree, degree.degreeToKey(Scale.major, 12)].debug("key, degree, back to key");
};

key, degree, back to key: [ 0, 0.0, 0.0 ]
key, degree, back to key: [ 1, 0.1, 1.0 ]
key, degree, back to key: [ 2, 1.0, 2.0 ]
key, degree, back to key: [ 3, 1.1, 3.0 ]
key, degree, back to key: [ 4, 2.0, 4.0 ]
key, degree, back to key: [ 5, 3.0, 5.0 ]
key, degree, back to key: [ 6, 3.1, 6.0 ]
key, degree, back to key: [ 7, 4.0, 7.0 ]
key, degree, back to key: [ 8, 4.1, 8.0 ]
key, degree, back to key: [ 9, 5.0, 9.0 ]
key, degree, back to key: [ 10, 5.1, 10.0 ]
key, degree, back to key: [ 11, 6.0, 11.0 ]
```

This algorithm always produces sharps. It would be more musically useful if, for example, key 10 in a major scale became a flat-7 (5.9) rather than a sharp-6 (5.1), but that's less critical than having a proper inverse.

**Comment by telephon** [comment]
Yes, sorry I think these many years ago I would have needed a thorough tester (I've never used it that much). We should maybe correct the current version.

@jamshark70 for your:
`degree = ((scale.detectIndex { |x| absKey < x }) ?? { scale.size }) - 1;`
it should be more efficient to use
`indexOfGreaterThan` (which I had written exactly for that purpose).
